### ISSUE TITLE
Batch Normalization train/test difference

### ISSUE THREAD

**Issue by shekhovt** [pytorch/pytorch#4253]
There is some difference in BN train/test behaviour using the exact statistics. 

```
from torch import Tensor
import torch.nn.functional as F
from torch.autograd import Variable

x = Tensor(1000,1)
x.uniform_()

m = Tensor(1)
v = Tensor(1)
eps = 0  # for simplicity
F.batch_norm(Variable(x), running_mean=m, running_var=v, momentum=1, training=True, eps=eps)
print((m - x.mean(dim=0)).abs().mean()) # gives 0.0
print((v - x.var(dim=0)).abs().mean()) # gives 0.0

# Eval version
y0 = F.batch_norm(Variable(x), running_mean=m, running_var=v, momentum=0, training=False, eps=eps)
# This is clear, it matches the formula
y0a = (x-m.view([1,-1]))/torch.sqrt(v.view([1,-1]))
print((y0.data-y0a).abs().mean()) # something around 1E-8

# Train version
y1 = F.batch_norm(Variable(x), running_mean=Tensor(1), running_var=Tensor(1), training=True, eps=eps)

print((y0-y1).abs().mean()) # something around 1E-4, not OK
```
Does not seem to be a numerical accuracy issue.
Maybe train version implements somewhat different formulae? If this is intended, could you please describe it in the docs? The difference is not big and may be not important asymptotically (like a biased variance estimator), but it amplifies through layers and breaks correctness tests of something I currently develop. At least I would like to know what's going on.
Thanks

**Comment by ngimel** [comment]
Yes, training and evaluation of batch norm use different formulae, and this is noted in the docs. http://pytorch.org/docs/0.3.0/nn.html#batchnorm2d

**Comment by shekhovt** [comment]
According to that the only difference is in mean and var estimates, but the formula is the same.
The example above is constructed to make the running mean and variance exactly match those of the batch. Yet there is a difference. Please reopen the issue.

**Comment by shekhovt** [comment]
Ok, I figured it out. For the running variance the unbiased estimator is used and for train-time variance the biased one. I suppose this becomes unimportant as long as there is more than one batch.
